File: 4. Median of Two Sorted Arrays/Sort and Pick/median_of_two_sorted_arrays.py
def get_median (array):
	len_array = len(array)
	even_length = (len_array % 2 == 0)
	median = None
	
	if even_length:
		median = (array[(len_array // 2) - 1] + array[len_array // 2]) / 2
	else:
		median = (array[(len_array // 2)])
		
	return median

#def get_combined_list (array_a, array_b):
def get_combined_list (array1, array2):
	#array1 and array2 have to be sorted for this to work
	
	array1.sort()
	array2.sort()
	
	len_array1 = len(array1)
	len_array2 = len(array2)
	
	array1_cursor = 0
	array2_cursor = 0
	
	out = []
	
	while (array1_cursor < len_array1) or (array2_cursor < len_array2):
		element_array1 = None
		element_array2 = None
		appending_element = None
		
		if (array1_cursor < len_array1):
			element_array1 = array1[array1_cursor]
		
		if (array2_cursor < len_array2):
			element_array2 = array2[array2_cursor]
		
		if (element_array1 != None and element_array2 != None):
			if element_array1 < element_array2:
				appending_element = element_array1
				array1_cursor += 1
			else:
				appending_element = element_array2
				array2_cursor += 1
		elif (element_array1 == None):
			appending_element = element_array2
			array2_cursor += 1
		elif (element_array2 == None):
			appending_element = element_array1
			array1_cursor += 1
			
		out.append(appending_element)
	
	return out
	
nums1 = [1,3]
nums2 = [2,4]

num_combined = get_combined_list(nums1, nums2)

print ("This is the handy work of Python.\n\n")
print (get_median(num_combined))
print ("\n\n\n")

File: 4. Median of Two Sorted Arrays/Sort and Pick/median_of_two_sorted_arrays.cs
using System;

namespace median_of_two_sorted_arrays_ns
{
	class median_of_two_sorted_arrays_c
	{
		static float get_median(int[] list){
			int len_list = list.Length;
			bool even_length = (len_list % 2 == 0);
			float nje;
			
			if (even_length){
				nje = ((float) list[len_list / 2] + (float) list[(len_list / 2) - 1]) / 2;
			} else {
				nje = list[len_list / 2];
			}
			
			return (nje);
		}
		
		static int[] combined_list (int [] list1, int[] list2){
			Array.Sort(list1);
			Array.Sort(list2);
			
			int len_list1 = list1.Length;
			int len_list2 = list2.Length;
			
			int[] nje = new int [len_list1 + len_list2];
			
			int nje_cursor = 0;
			int list1_cursor = 0;
			int list2_cursor = 0;
			
			while ((list1_cursor < len_list1) || (list2_cursor < len_list2)){
				int? element_list1 = null;
				int? element_list2 = null;
				int? appending_element = null;
				
				if (list1_cursor < len_list1){
					element_list1 = list1[list1_cursor];
				}
				
				if (list2_cursor < len_list2){
					element_list2 = list2[list2_cursor];
				}
				
				if ((element_list1 != null) && (element_list2 != null)){
					if (element_list1 < element_list2){
						appending_element = element_list1;
						list1_cursor += 1;
					} else {
						appending_element = element_list2;
						list2_cursor += 1;
					}
				} else if (element_list1 == null){
					appending_element = element_list2;
					list2_cursor += 1;
				} else if (element_list2 == null){
					appending_element = element_list1;
					list1_cursor += 1;
				}
				if (appending_element != null){
					nje[nje_cursor] = appending_element ?? -222;
				}
				nje_cursor += 1;
			}
			
			return (nje);
		}
		
		static void Main (String [] args){
			int [] list1 = {1,3};
			int [] list2 = {2,4};
			int [] list_combined = combined_list(list1, list2);
			
			float median = get_median (list_combined);
			
			Console.WriteLine("This is the handy work of C#\n\n\n");
			Console.WriteLine(median);
			Console.WriteLine("\n\n\n");
		}
	}
}
